Steque.java

```diff
@@ -30,63 +32,133 @@
  *
  */
 public class Steque<Item> implements Iterable<Item> {
-
+    private Node first, last;
+    private class Node{
+        private Item item;
+        private Node next;
+    }
+    
 
     /**
      * constructs a steque object.
      */
     public Steque() {
-
+        first = last = null;
+        
     }
     
     
     /**
      * inserts an item in the steque in queue fashion.
      * @param item Item to be inserted.
      */
+    // time complexity: O(1) ,space complexity: O(n)
     public void enqueue(Item item) {
```

Comment:
In one branch give only that particular PR, but here you have written entire code in a single PR.